metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DepublishedRecordDao.java

```diff
@@ -147,59 +150,107 @@ long countDepublishedRecordsForDataset(String datasetId) {
+   * It is mainly used for minimize internal to the application database requests.</p>
    *
-   * @param datasetId The dataset for which to do this.
+   * @param datasetId The dataset for which to retrieve the records. Cannot be null.
+   * @param sortField The sorting field. Cannot be null.
+   * @param sortDirection The sorting direction. Cannot be null.
+   * @param depublicationState The depublication state of the records. Can be null.
+   * @param searchQuery Search query for the record ID. Can be null.
+   * @return A (possibly empty) list of depublished records.
    */
-  public void markAllAsNotDepublished(String datasetId){
+  public Set<String> getAllDepublishedRecords(String datasetId,
+      DepublishedRecordSortField sortField, SortDirection sortDirection,
+      DepublicationStatus depublicationState, String searchQuery) {
+    final Query<DepublishedRecord> query = prepareQueryForDepublishedRecords(datasetId, sortField,
+        sortDirection, depublicationState, searchQuery);
+    query.project(DepublishedRecord.RECORD_ID_FIELD, true);
+    query.project(DepublishedRecord.ID_FIELD, false);
+
+    // Execute query with correct pagination
+    final List<DepublishedRecord> result = getListOfQuery(query, null);
+
+    // Convert result to right object.
+    return result.stream().map(DepublishedRecord::getRecordId).collect(Collectors.toSet());
+  }
 
+  private Query<DepublishedRecord> prepareQueryForDepublishedRecords(String datasetId,
+      DepublishedRecordSortField sortField, SortDirection sortDirection,
+      DepublicationStatus depublicationState, String searchQuery) {
     // Create query.
     final Query<DepublishedRecord> query = morphiaDatastoreProvider.getDatastore()
-            .createQuery(DepublishedRecord.class);
+        .createQuery(DepublishedRecord.class);
     query.field(DepublishedRecord.DATASET_ID_FIELD).equal(datasetId);
+    if (Objects.nonNull(depublicationState)) {
+      query.field(DepublishedRecord.DEPUBLICATION_STATE_FIELD).equal(depublicationState);
+    }
+    if (StringUtils.isNotBlank(searchQuery)) {
+      query.field(DepublishedRecord.RECORD_ID_FIELD).contains(searchQuery);
+    }
+
+    // Set ordering
+    query.order(sortDirection.createSort(sortField.getDatabaseField()));
+    return query;
+  }
+
+  /**
+   * This method marks record ids with the provided {@link DepublicationStatus} and {@link Date}
+   * where appropriate.
+   * <p>A {@link DepublicationStatus#PENDING_DEPUBLICATION} unsets the depublication date</p>
+   * <p>A {@link DepublicationStatus#DEPUBLISHED} sets the depublication date with the one
+   * provided</p>
+   *
+   * @param datasetId the dataset for which to do this
+   * @param depublicationStatus the depublication status. Null status is translated to {@link
+   * DepublicationStatus#PENDING_DEPUBLICATION}
+   * @param depublicationDate the depublication date. Can be null
+   */
+  public void markRecordIdsWithDepublicationStatus(String datasetId,
```

Comment:
Should you bring this method back in the original state? I think this will only ever be called with DepublicationStatus.PENDING_DEPUBLICATION (see my remark in the WorkflowPostProcessor. Or do you plan to make this method generic enough to cover the other use case as well?

Reply:
I guess we could keep it and extended with the record ids to match for other states?
Shouldn't harm the execution of the query if we send it as a batch which we usually do

Reply:
OK, doesn't matter too much.